apps/emqx_conf/src/emqx_conf_app.erl

```diff
@@ -327,3 +327,23 @@ has_deprecated_file(#{conf := Conf} = Info) ->
             %% Conf is not empty if deprecated file is found.
             Conf =/= #{}
     end.
+
+traverse_and_collect_files(DataDir) ->
+    SubDirs = lists:map(fun(D) -> filename:join(DataDir, D) end, ?DATA_DIRS),
+    do_traverse_and_collect_files(SubDirs, DataDir, _Acc = []).
+
+do_traverse_and_collect_files([] = _SubDirs, _DataDir, Acc) ->
+    Acc;
+do_traverse_and_collect_files([SubDir | Rest], DataDir, Acc0) ->
+    %% This function already drops any non-regular file, including symlinks.
+    Acc = filelib:fold_files(
+        SubDir,
+        _Regex = "",
+        _Recursive = true,
+        fun(Path0, Acc) ->
+            Path = lists:flatten(string:replace(Path0, DataDir ++ "/", "", leading)),
```

Comment:
can this happen? `DataDir` already has a trailing `/`

Reply:
maybe make a help function
`to_data_dir_relative_path`

Reply:
not always :thinking: 

```elixir
iex(emqx@127.0.0.1)2> :emqx.data_dir
~c"data"
```

Reply:
:checkered_flag: